lib/pbio/drv/bluetooth/bluetooth_stm32_cc2640.c

```diff
@@ -716,6 +724,11 @@ static PT_THREAD(scan_and_connect_task(struct pt *pt, pbio_task_t *task)) {
         }
 
         device_discovery_done = false;
+
+        PROCESS_CONTEXT_BEGIN(&pbdrv_bluetooth_spi_process);
+        etimer_set(&observe_restart_timer, OBSERVE_RESTART_INTERVAL);
+        PROCESS_CONTEXT_END(&pbdrv_bluetooth_spi_process);
+        observe_restart_enabled = true;
```

Comment:
Would it be better to move this to the end of this task, just before passive scanning is restarted again?